packages/amplify-graphql-relational-transformer/src/schema.ts

```diff
@@ -226,6 +226,20 @@ export const ensureHasOneConnectionField = (config: HasOneDirectiveConfiguration
     );
   }
 
+  const subscriptionFilterInputName = getSubscriptionFilterInputName(object.name.value);
+  const filterSubscriptionInput = ctx.output.getType(subscriptionFilterInputName) as InputObjectTypeDefinitionNode;
+  if (filterSubscriptionInput) {
+    updateFilterConnectionInputWithConnectionFields(
```

Comment:
It seems that this function is generating the `ModelStringType` instead of `ModelSubscriptionStringType`

Reply:
Good catch, I'll work on fixing this.